### ISSUE TITLE
Overriding `DataFetcherExceptionHandler` has no effect

### ISSUE THREAD

**Issue by lewimuchiri** [yahoo/elide#2954]
As a follow-up to issue [#2912](https://github.com/yahoo/elide/issues/2912), I still find that when you provide your own `DataFetcherExceptionHandler` Bean, Elide still falls back to the default `SimpleDataFetcherExceptionHandler`. This is in version 7.0.0-pr3. 
I have seen this comment in the AsyncExecutorService and was wondering "why though?":
<img width="511" alt="image" src="https://user-images.githubusercontent.com/1880626/233780204-45ca01ca-8176-4a3d-a0f3-be667afc77a3.png">

I had hoped that commit [#2931](https://github.com/yahoo/elide/pull/2931) would enable me to provide my own exception handler implementation (since the default GraphQL one has this annoying `"Exception while fetching data"` prefix (see screenshot)
<img width="401" alt="image" src="https://user-images.githubusercontent.com/1880626/233780394-09d06483-8c1d-4cac-9b4c-a9556c3b06db.png">

## Expected Behavior
The custom `DataFetcherExceptionHandler` should be used as opposed to the default `SimpleDataFetcherExceptionHandler`.
```java
    @Bean
    @Primary
    public DataFetcherExceptionHandler getGraphQLExceptionHandler() {
        return new CustomExceptionHandler();
    }
```

**Comment by aklish** [comment]
Are you using Async?  I looked at the code again and I believe the comment
might be in error.  However, it should only affect async queries.   There
was not a clean way to plumb the ExceptionHandler to async (possible - just
hacky).  If you are not using the Async API, then something else is wrong.

**Comment by lewimuchiri** [comment]
No, I'm not using Async. Was wondering about that too, but no, I'm not using the Async API.
For sure there must be another problem elsewhere because I provided my own `DataFetcherExceptionHandler` and even the `QueryRunners` bean and none of them worked. The default `SimpleDataFetcherExceptionHandler` is used

**Comment by justin-tay** [comment]
This is just a guess but could you try to change

https://github.com/yahoo/elide/blob/6ba885ea84e10374930a563454c0c2d37b1a76e8/elide-graphql/src/main/java/com/yahoo/elide/graphql/QueryRunner.java#L102-L104

to 

```java
api = GraphQL.newGraphQL(builder.build())
                .defaultDataFetcherExceptionHandler(exceptionHandler)
                .build();
```

If you can't build a snapshot of Elide you probably need to do some copy and paste of `QueryRunners` and `QueryRunner`.

**Comment by aklish** [comment]
Just released @justin-tay fix in 7.0.0-pr4

**Comment by lewimuchiri** [comment]
Yes, this works fine now
